octree_creation_app/utils.py

```diff
@@ -61,13 +61,15 @@ def create_octree_from_octrees(meshes: list[Octree | TreeMesh]) -> TreeMesh:
         if isinstance(mesh, Octree) and mesh.octree_cells is not None:
             centers = mesh.centroids
             levels = treemesh.max_level - np.log2(mesh.octree_cells["NCells"])
-        elif isinstance(mesh, TreeMesh):
+        elif isinstance(mesh, TreeMesh) and mesh.cell_centers is not None:
             centers = mesh.cell_centers
             levels = (
                 treemesh.max_level
                 - mesh.max_level
                 + mesh.cell_levels_by_index(np.arange(mesh.nC))
             )
+        else:
+            raise TypeError(f"All meshes must be Octree or TreeMesh, not {type(mesh)}")
```

Comment:
...and must have octree_cells set for octree and cell_centers for treemesh.

Reply:
done